### DISCUSSION TITLE
Move Files upon Completion

### DISCUSSION THREAD

**Discussion by swiwchar** [Yutaka-Sawada/MultiPar#67] [Ideas]
Could you possibly add an option that upon successful completion of recovery, move all related files (recovery and data files) to a new directory?
I have many recovery sets in one directory and manually moving/testing is very cumbersome.

**Comment by Yutaka-Sawada** [comment]
I don't want to make an option for rare usage. I made an option to use file-list in Batch scripting.
"Automated tasks" tab -> "Batch Processing after Verification or Repair" section -> "Use list of files in script" check-box

Check both "Start batch script" and "Use list of files in script". In "save" folder, "chk_list.tmp" and "src_list.tmp" are made. They are list of PAR files and Souce files. Batch script can refer them. I wrote sample script to make a new fodler and move files in. 

Because MultiPar doesn't delete them automatically, you need to erase the lists after your usage. Batch script may erase them.

I put the sample (gui_sample_2022-07-03.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOg0cF2UHcs709Icv4?e=UUj7bE). You may test the new option and batch scripting feature are good or not enough. Because it's a sample, the feature may not be shown in normal release version. I'm lazy to update language UI DLLs.

**Comment by swiwchar** [comment]
Thank you, this worked wonderfully.
Had to add a couple of double-quotes for mkdir.

Very nice to see you continue to develop this.

**Comment by Yutaka-Sawada** [comment]
I plan to save more detailed status information on a file. I test JSON format for Python script now. It can store multiples items in a single file. It's better than using some temporary files. You may read JSON file on Batch script by using PowerShell.

Do you use par2j64.exe or MultiPar.exe ? MultiPar.exe will save common items between PAR1/PAR2/SFV/MD5. par2j64.exe on Command Prompt will save more items.

**Comment by Yutaka-Sawada** [comment]
I made a sample to save JSON file after repair. The GUI (MultiPar.exe) saves JSON file, when all source files are complete after verification or repair. So, there is no status information (like; damaged, missing, or misnamed). The client (par2j64.exe) saves JSON file after verification, even when source files aren't complete. So, it stores some information of each file's status. I don't know which is useful for you.

I put two samples (json_sample_2022-10-14.zip and gui_sample_2022-10-15.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOg0cF2UHcs709Icv4?e=UUj7bE). I includes a sample Python script, too. Please test them.

Because this is a sample for test, the feature may not be shown in normal release version. Because I'm lazy to implement a special feature for rare case, I recommend to use Python script. By calling Python script, I will be able to remove two items from Batch Processing in future; "Open a source file" and "Expand an archive file (by 7-Zip)". But, I don't know how many users will try Python script.
